assets/js/blocks/add-to-cart-form/block.json

```diff
@@ -0,0 +1,11 @@
+{
+	"name": "woocommerce/add-to-cart-form",
+	"version": "1.0.0",
+	"title": "Add to Cart form",
```

Comment:
The more I read this, the more I'm inclined to keep the title as Add-to-cart button rather than Add-to-cart form: at the end of the day, it's a button with some extra content displayed, depending on the product type. What do you think @vivialice and @Aljullu ?

Reply:
> The more I read this, the more I'm inclined to keep the title as Add-to-cart button rather than Add-to-cart form

I don't have a strong opinion about the block name. But I do think we should avoid having two different blocks with the same name, so I would like if we can look for an alternative to "Add to cart button".

> at the end of the day, it's a button with some extra content displayed, depending on the product type

While this is true for simple products, there are many extensions which add extra fields to this block, ie: Product Add-Ons or Bookings. In those cases, naming this block "Add to cart button" might be confusing (ie: Bookings renders an entire calendar if I'm not wrong).

Reply:
It seems the ultimate action is to add the product to the cart, but the product can also be customized with this block, I was wondering if "Product Customization form" would be a good name 🤔

Reply:
> I don't have a strong opinion about the block name. But I do think we should avoid having two different blocks with the same name, so I would like if we can look for an alternative to "Add to cart button". While this is true for simple products, there are many extensions which add extra fields to this block, ie: Product Add-Ons or Bookings. In those cases, naming this block "Add to cart button" might be confusing (ie: Bookings renders an entire calendar if I'm not wrong).

Hm yeah, good point, thanks!

> I was wondering if "Product Customization form" or "Product Customization form for Cart" to be more specific, would be good names 🤔

Yeah, I'm not sure either! Maybe we can group these ideas into another issue and discuss them with Viv & the team later on.

Reply:
Just chiming in as someone with 2 plugins that add custom form fields to the current add to cart form. The `[add_to_cart]`  shortcode equivalent is currently only a button, but that's a shortcoming as you cannot use it on complex product types (using Name Your Price, Product Addons, Mix and Match, Product Bundles, etc). I had to write an `[add_to_cart_form]` shortcode to get parity. 

So my opinion (that nobody asked for :) ) is that you might want to call it "add to cart" but  _always_ display the entire form? For a simple product with no add-ons that would not be much more than the form, quantity, and add to cart button.. so would not appear/function much differently from "just a button". For other, more complex types, there's no value to a standalone button as it cannot function without the rest of it's form fields.

Reply:
Hello @helgatheviking ! Thank you so much for your contribution to this discussion; we appreciate your input here!

> So my opinion (that nobody asked for :) ) is that you might want to call it "add to cart" but always display the entire form? For a simple product with no add-ons that would not be much more than the form, quantity, and add to cart button.. so would not appear/function much differently from "just a button". For other, more complex types, there's no value to a standalone button as it cannot function without the rest of it's form fields.

IMO this is a great idea. Paging @vivialice here for her input.

Reply:
I just remembered @vivialice is probably AFK for a design meetup, and we need to progress other work depending on this block, so I'm going ahead and opening a separate issue to continue the discussion regarding the naming here (we still have some time until the release). Thanks again for your input @helgatheviking !

Reply:
Here's the new issue for this discussion: https://github.com/woocommerce/woocommerce-blocks/issues/8453

Reply:
Thanks for your feedback, @helgatheviking! To provide some screenshots, this is the Add to Cart button and Add to Cart form (or however ends up being named :smile: ) of a variable product:

Add to Cart button | Add to Cart form
--- | ---
<img src="https://user-images.githubusercontent.com/3616980/219339326-c512d115-ce12-4e45-b566-cf1883a19491.png" alt="" width="368" /> | <img src="https://user-images.githubusercontent.com/3616980/219339397-a5fa683c-3505-438f-83ed-928652927f61.png" alt="" width="425" />

They serve two different purposes, the button is useful in product grids and contexts where the entire form would crowd the layout too much, while the form is useful in contexts where the customer is expected to finalize the selection and add the product to the cart.

Currently, the _Add to cart button_ is only available inside the Products block and the _Add to cart form_ will only be available inside the Single Product template. So right now, there is no confusion between the two. But in the future, we might introduce a _Product_ block which will allow showcasing products in the front page, for example. In that case, I envision there might be use cases for both blocks: some stores might want to add the _Add to cart button_ block so customers get redirected to the Single Product page (assuming it's not a simple product), and some other stores might want to add the _Add to cart form_ block, so customers can choose between the options right in the front page without having to visit the Single Product page.

That's why I think it's important for us to make a clear distinction of the two blocks. I don't have strong opinions about the specific names, but as I mentioned in my previous comment, I do think they need different names. I'm also inclined to avoid the _Add to Cart_ name for one of them because I think it will make it difficult to distinguish which of the two blocks it is.